source/MaterialXGenShader/Util.cpp

```diff
@@ -608,4 +608,85 @@ void findRenderableElements(const DocumentPtr& doc, std::vector<TypedElementPtr>
     }
 }
 
+bool getUIProperties(const ValueElementPtr nodeDefElement, UIProperties& uiProperties)
```

Comment:
This are the utilities to extract out UI properites from an element path. At some point these have to be removed from here to proper places (in a run-time lib?) as it's just a grab bag of stuff.